core/client_test.go

```diff
@@ -0,0 +1,138 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/yomorun/yomo/core/frame"
+	"github.com/yomorun/yomo/core/metadata"
+	"github.com/yomorun/yomo/core/router"
+	"github.com/yomorun/yomo/pkg/config"
+	"github.com/yomorun/yomo/pkg/logger"
+)
+
+const testaddr = "127.0.0.1:19999"
+
+func TestClientDialNothing(t *testing.T) {
+	ctx := context.Background()
+
+	client := NewClient("source", ClientTypeSource)
+
+	assert.Equal(t, ConnStateReady, client.State(), "client state should be ConnStateReady")
+
+	err := client.Connect(ctx, testaddr)
+
+	assert.Equal(t, ConnStateDisconnected, client.State(), "client state should be ConnStateDisconnected")
+
+	qerr := &quic.IdleTimeoutError{}
+	assert.ErrorAs(t, err, &qerr, "dial must timeout")
```

Comment:
this test seems meaningless?

Reply:
It makes meaning for testing state `ConnStateDisconnected`.